platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -166,11 +166,15 @@ private static Optional<String> diagnosticCount(Log logger, String kind, int num
 
     @VisibleForTesting
     void buildProblem(Diagnostic<? extends JavaFileObject> diagnostic, ProblemSpec spec) {
-        spec.severity(mapKindToSeverity(diagnostic.getKind()));
+        Severity severity = mapKindToSeverity(diagnostic.getKind());
+        spec.severity(severity);
         addId(spec, diagnostic);
         addFormattedMessage(spec, diagnostic);
         addDetails(spec, diagnostic);
         addLocations(spec, diagnostic);
+        if (severity == Severity.ERROR) {
+            spec.solution(CompilationFailedException.RESOLUTION_MESSAGE);
```

Comment:
🤔 Why add something in here and also have an if-else in [CompilationFailedException.java](https://github.com/gradle/gradle/pull/30866/files#diff-fd2c5ff64ffab2a216ee2f9fd0952952bfbb3a1a831b82c147876a5cdad50430)?

Reply:
From my (perhaps limited) understanding:
* If there are compilation warnings, they may appear as additional problems. Those should not have a dedicated solution at the moment.
* The logic in `CompilationFailedException` is temporary IMO but has to be there to cover the cases where the exception is created by one of the constructors not passing in `Problem`s.

Reply:
I agree the code in `CompilationFailedException` should be temporary.